### ISSUE TITLE
Tectia (sftpg3) does not like attribute format

### ISSUE THREAD

**Issue by RomanHargrave** [proftpd/proftpd#762]
A user is having a hard time connecting with Tectia SSH G3. Upon successful authentication they are presented with

```
Error: Could not open connection to `user@server': Bad attrs reply
```

This issue has come up on the forums before, where it was suggested that `FSOptions IgnoreExtendedAttributes` be enabled; however, this did not work at that time, and it does not work for me. Ultimately the solution was to recompile without extended attribute support; however this is really something that should be (properly) configurable.

### ProFTPD Version and Configuration

Version Info (***please note that the build date is patched to May 1 for Debian's reproducible builds***)
```
Compile-time Settings:
  Version: 1.3.6 (stable)
  Platform: LINUX [Linux 4.15.0-36-generic x86_64]
  Built: Tue May 1 2018 12:57:41 UTC
  Built With:
    configure  '--build=x86_64-linux-gnu' '--includedir=${prefix}/include' '--mandir=${prefix}/share/man' '--infodir=${prefix}/share/info' '--sysconfdir=/etc' '--localstatedir=/var' '--disable-silent-rules' '--libdir=${prefix}/lib/x86_64-linux-gnu' '--libexecdir=${prefix}/lib/x86_64-linux-gnu' '--disable-maintainer-mode' '--disable-dependency-tracking' '--prefix=/usr' '--with-includes=/usr/include/postgresql:/usr/include/mysql' '--mandir=/usr/share/man' '--sysconfdir=/etc/proftpd' '--localstatedir=/run' '--libexecdir=/usr/lib/proftpd' '--enable-sendfile' '--enable-facl' '--enable-dso' '--enable-autoshadow' '--enable-ctrls' '--enable-ipv6' '--enable-nls' '--enable-memcache' '--with-lastlog=/var/log/lastlog' '--enable-pcre' '--disable-strip' '--enable-redis' '--build' 'x86_64-linux-gnu' '--with-shared=mod_unique_id:mod_site_misc:mod_load:mod_ban:mod_quotatab:mod_sql:mod_sql_mysql:mod_sql_postgres:mod_sql_sqlite:mod_sql_odbc:mod_dynmasq:mod_quotatab_sql:mod_ldap:mod_quotatab_ldap:mod_ratio:mod_tls:mod_rewrite:mod_radius:mod_wrap:mod_wrap2:mod_wrap2_file:mod_wrap2_sql:mod_quotatab_file:mod_quotatab_radius:mod_facl:mod_ctrls_admin:mod_copy:mod_deflate:mod_ifversion:mod_geoip:mod_exec:mod_sftp:mod_sftp_pam:mod_sftp_sql:mod_shaper:mod_sql_passwd:mod_ifsession:mod_auth_otp:mod_tls_redis:mod_wrap2_redis:mod_redis:mod_memcache:mod_tls_memcache:mod_readme:mod_snmp' 'build_alias=x86_64-linux-gnu' 'CFLAGS=-g -O2 -fdebug-prefix-map=/tmp/src=. -fstack-protector-strong -Wformat -Werror=format-security' 'LDFLAGS=-Wl,-z,relro' 'CPPFLAGS=-Wdate-time -D_FORTIFY_SOURCE=2' 'CXXFLAGS=-g -O2 -fdebug-prefix-map=/tmp/src=. -fstack-protector-strong -Wformat -Werror=format-security'

  CFLAGS: -g2 -g -O2 -fdebug-prefix-map=/tmp/src=. -fstack-protector-strong -Wformat -Werror=format-security -Wall -fno-omit-frame-pointer -Werror=implicit-function-declaration
  LDFLAGS: -L$(top_srcdir)/lib -Wl,-z,relro -rdynamic  -L/usr/lib/x86_64-linux-gnu -L/usr/lib/x86_64-linux-gnu
  LIBS: -lacl  -lpcreposix -lpcre -lssl -lcrypto -lcap  -lpam -lsupp -lattr -lnsl -lresolv -lresolv -lcrypt -ldl -lhiredis -lmemcachedutil -lmemcached 

  Files:
    Configuration File:
      /etc/proftpd/proftpd.conf
    Pid File:
      /run/proftpd.pid
    Scoreboard File:
      /run/proftpd.scoreboard
    Header Directory:
      /usr/include/proftpd
    Shared Module Directory:
      /usr/lib/proftpd

  Info:
    + Max supported UID: 4294967295
    + Max supported GID: 4294967295

  Features:
    + Autoshadow support
    + Controls support
    + curses support
    - Developer support
    + DSO support
    + IPv6 support
    + Largefile support
    + Lastlog support
    + Memcache support
    + ncursesw support
    + NLS support
    + Redis support
    - Sodium support
    + OpenSSL support
    + PCRE support
    + POSIX ACL support
    + Shadow file support
    + Sendfile support
    + Trace support
    + xattr support

  Tunable Options:
    PR_TUNABLE_BUFFER_SIZE = 1024
    PR_TUNABLE_DEFAULT_RCVBUFSZ = 8192
    PR_TUNABLE_DEFAULT_SNDBUFSZ = 8192
    PR_TUNABLE_ENV_MAX = 2048
    PR_TUNABLE_GLOBBING_MAX_MATCHES = 100000
    PR_TUNABLE_GLOBBING_MAX_RECURSION = 8
    PR_TUNABLE_HASH_TABLE_SIZE = 40
    PR_TUNABLE_LOGIN_MAX = 256
    PR_TUNABLE_NEW_POOL_SIZE = 512
    PR_TUNABLE_PATH_MAX = 4096
    PR_TUNABLE_SCOREBOARD_BUFFER_SIZE = 80
    PR_TUNABLE_SCOREBOARD_SCRUB_TIMER = 30
    PR_TUNABLE_SELECT_TIMEOUT = 30
    PR_TUNABLE_TIMEOUTIDENT = 10
    PR_TUNABLE_TIMEOUTIDLE = 600
    PR_TUNABLE_TIMEOUTLINGER = 10
    PR_TUNABLE_TIMEOUTLOGIN = 300
    PR_TUNABLE_TIMEOUTNOXFER = 300
    PR_TUNABLE_TIMEOUTSTALLED = 3600
    PR_TUNABLE_XFER_SCOREBOARD_UPDATES = 10
```
```
# SFTP-Only ProFTPD Configuration
# Runs on 22/TCP

ModulePath /usr/lib/proftpd
LoadModule mod_sql.c
LoadModule mod_sql_sqlite.c
LoadModule mod_sql_crypt_r.c
LoadModule mod_sftp.c
LoadModule mod_sftp_sql.c
LoadModule mod_delay.c

# Set off to disable IPv6 support which is annoying on IPv4 only boxes.
UseIPv6              on
IdentLookups         off

ServerName           "FTPD"
ServerType           standalone
ScoreboardFile       /var/sftp/scoreboard
PidFile              /var/sftp/proftpd.pid
DeferWelcome         off

MultilineRFC2228     on
DefaultServer        on
ShowSymlinks         on

TimeoutNoTransfer    600
TimeoutStalled       600
TimeoutIdle          1200

DisplayChdir         .message true
ListOptions          "-l"

DenyFilter           \*.*/

CreateHome on

RootLogin on

# SQL Queries
<IfModule mod_sql.c>
   SQLbackend     sqlite3
   SQLConnectInfo /path/to/users.db

   # Not used for connecting to the database, 
   # rather, specifies the format in which passwords are stored
   SQLAuthenticate users
   SQLAuthTypes crypt_r

   SQLUserInfo users username password NULL NULL directory NULL

   SQLNamedQuery get-user-authorized-keys SELECT "..."
</IfModule>

# Provide an SFTP server
<IfModule mod_sftp.c>
   # Store SFTP Keys in DB
   # debian ships this module.
   <IfModule mod_sftp_sql.c>
      SFTPAuthorizedUserKeys sql:/get-user-authorized-keys
   </IfModule>

   SFTPEngine        on
   Port              22
   SFTPHostKey       /path/to/ssh_host_dsa_key
   SFTPHostKey       /path/to/ssh_host_rsa_key
   SFTPAuthMethods   publickey password

   # Mandate a fixed window size for noncompliant SSH clients that request invalid window sizes
   # See http://www.proftpd.org/docs/contrib/mod_sftp.html#SFTPBadWindowSize
   SFTPClientMatch ".*WS_FTP.*" channelWindowSize 256MB
   SFTPClientMatch ".*XFB.*"    channelWindowSize 1GB
   SFTPClientMatch ".*XFB.*"    sftpUTF8ProtocolVersion 3
</IfModule>

# sticks all users in their home directories
DefaultRoot ~

# using DB auth, no underlying system users and thus no shell provisions
RequireValidShell off

# Port directive disabled as standard FTP is not in use
# Port 21

# To prevent DoS attacks, set the maximum number of child processes
# to 30.  If you need to allow more than 30 concurrent connections
# at once, simply increase this value.  Note that this ONLY works
# in standalone mode, in inetd mode you should use an inetd server
# that allows you to limit maximum number of processes per service
# (such as xinetd)
MaxInstances   30

# Set the user and group that the server normally runs at.
User  proftpd
Group nogroup

# Umask 022 is a good standard umask to prevent new files and dirs
# (second parm) from being group and world writable.
Umask 022  022
AllowOverwrite on

TransferLog /path/to/xferlog

# Delay engine reduces impact of the so-called Timing Attack described in
# http://www.securityfocus.com/bid/11430/discuss
# It is on by default. 
<IfModule mod_delay.c>
   DelayEngine on

   # The server uses this file to store delay timing statistics
   # and automatically adjust to traffic
   DelayTable /path/to/delay_table

   # Set the default failed login delay time to Ns/attempt
   DelayOnEvent FailedLogin 2s
</IfModule>
```

**Comment by RomanHargrave** [comment]
This issue is a re-emergence of #593

I am going to get in touch with the Debian ProFTPd maintainers and see if this fix is present in their distribution.

**Comment by RomanHargrave** [comment]
This is a debian issue.

**Comment by Castaglia** [comment]
Do you happen to have a link/reference to the Debian-specific cause for this, for future reference?  Thanks!

**Comment by RomanHargrave** [comment]
@Castaglia 

The bug report: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=913824

The source code for the debian ProFTPd package: https://salsa.debian.org/debian-proftpd-team/proftpd/tree/master

Hilmar or someone else should get in and add the patch pretty quickly. The last time around (with #710) things got backported pretty fast.

***With that being said*** fixes from #710 are present in the debian **build** but the latest package is still updated. Currently, the package is marked for removal in december because the debian bug for #710 is marked critical and unresolved.

I believe this issue affects Ubuntu users, by extension.
